### DISCUSSION TITLE
Expectation values differ widely between RWA and lab frame

### DISCUSSION THREAD

**Discussion by khanhuyengiang** [qutip/qutip#2559] [General]
I am currently trying to simulate the Hamiltonian from this paper https://arxiv.org/pdf/[2106.00675](https://arxiv.org/pdf/2106.00675), specifically equation (4) and equation (5):

Equation (4) - Duffing oscillator:
$$H_{siZZle}/h = \sum_{i=\{0,1\}} (\nu_i \hat a_i^\dagger \hat a_i + \frac{\alpha_i}{2} \hat a_i^\dagger \hat a_i(\hat a_i^\dagger \hat a_i-1) + J(\hat a_0^\dagger + \hat a_0)(\hat a_1^\dagger + \hat a_1) + \sum_{i=\{0,1\}}\Omega_i \cos(2\pi\nu_d t + \phi_i) \left( a_i^\dagger + a_i \right) $$

and equation (5) the dressed RWA Hamiltonian:
$$H_eff/h = \tilde\nu_{ZI} ZI/4 + \tilde\nu_{IZ}  IZ/4 + \tilde\nu_{ZZ}  ZZ/4$$

I use the parameters given in the paper

```
# MHz
v_0 = 4953   # Frequency of qubit A
v_1 = 5014  # Frequency of qubit B
alpha_0 = -276  # Anharmonicity of qubit A
alpha_1 = -286  # Anharmonicity of qubit B
J = 7.745  # Coupling strength between qubits A and B
Omega_0 = 37.5
Omega_1 = 15
v_d = 5075
phi_0 = 0
phi_1 = 0
```

However, I received very different plots when I use `mesolve` on these 2 Hamiltonians:
![image](https://github.com/user-attachments/assets/88b2a6a7-6a44-49aa-80d0-28232a268b78)

![image](https://github.com/user-attachments/assets/14aa5332-8415-4070-9995-900271061f3b)

I want to ask where the discrepencies come from and how to resolve this. Many thanks!

**Comment by BoxiLi** [comment]
What you are observing is exactly the rotating wave approximation. In the first plot, the observables undergo fast oscillation with ~5GHz. You have to transfer your saved states into the rotating frame to match them using `(-1.j * H_idle * t).expm()) @ state_t` where H_idle is the time-independent part in equation 4.
